bgcval2/analysis_timeseries.py

```diff
@@ -139,6 +163,180 @@ def build_list_of_suite_keys(suites, debug=True):
+        func = std_functions[functionname]
+
+    if functionname.find(':') > -1:
+        [functionFileName,functionname] = functionname.split(':')
+        lst = functionFileName.replace('.py','').replace('/', '.').split('.')
+        modulename =  '.'.join(lst)
+
+        print("parseFunction:\tAttempting to load the function:",functionname, "from the:",modulename)
+        mod = __import__(modulename, fromlist=[functionname,])
+        func = getattr(mod, functionname)
+    return func     
+
+
+def load_function_kwargs(dictionary, m_or_d):
+    """
+    Loads key word arguments from yaml file dictionairy.
+    """
+    data_kwargs = {}
+    #, data_convert_kwargss = [], []
+    #####
+    # Looking for kwargs to pass to convert:
+    for key, value in dictionary.items():
+        searchFor =  m_or_d+'_convert_'
+        findstr = key.find(searchFor)
+        if findstr==-1: continue
+        kwargkey = key[len(searchFor):]
+        data_kwargs[kwargkey] = value
+    return data_kwargs 
+
+
+def findReplaceFlag(filepath, flag, new_value):
+    """
+    Looking for $FLAG in the filepath.
+    If found, we replace $FLAG with new_value
+    """
+    lookingFor = '$'+flag.upper()
+    if filepath.find(lookingFor) ==-1: 
+        return filepath
+    filepath = filepath.replace(lookingFor, new_value)
+    return filepath
+
+
+def parse_list_from_string(list1):
+    """
+    This tool takes a string and returns it as a list.
+    """
+    if isinstance(list1, list):
+       return list1
+
+    list1 = list1.replace('\t', ' ')
+    list1 = list1.replace(',', ' ')
+    list1 = list1.replace('  ', ' ')
+    list1 = list1.replace('\'', '')
+    list1 = list1.replace('\"', '')
+    list1 = list1.replace(';', '')
+    while list1.count('  ')>0:
+        list1 = list1.replace('  ', ' ')
+    #if len(list1)==0: return []
+    return list1.split(' ')
+
+
+def load_key_file(key, paths, jobID):
+    """
+    load_key_file:
+        takes an input file from key_files directory
+        and outputs a dictionairy.
+    """
+    output_dict = {}
+    print('load_key_file: checking {key}', key)
+    key_yml_path = os.path.join(os.path.join(paths.bgcval2_repo, 'key_files', ''.join([key.lower(),'.yml'])))
+    if os.path.exists(key_yml_path):
+        print('key_yml_path exists:', key_yml_path)
+    else:
+        print('key_yml_path does not exist:', key_yml_path)
+        raise FileNotFoundError
+
+    # Open yml file:
+    with open(key_yml_path, 'r') as openfile:
+        dictionary = yaml.safe_load(openfile)
+
+    if not dictionary or not isinstance(dictionary, dict):
+        print(f"Key Yaml file {key_yml_path} "
+              "is either empty or corrupt, please check its contents")
+        sys.exit(1)
+
+    # add jobID to dict:
+    dictionary['jobID'] = jobID
+
+    # Load basic fields: 
+    for field in ['name', 'units', 'dimensions', 'modelgrid']:
+        output_dict[field] = dictionary.get(field, None)
```

Comment:
could it be that one or more of those `field` in list are not in `dictionary`? In that case you're setting a key with `None` value, which is badd

Reply:
I think they're pretty important... Might be able to get away without modelgrid for some analyses, just about. And it might be possible to determine the number of dimensions from the data post hoc.

Reply:
OK but my worry was a purely technical one: `output_dict[field]` could be set to `None` if `dictionary` doesn't have it as key - but you know better what's the contents of stuff there